astronet/direct_tensor/predict.py

```diff
@@ -0,0 +1,382 @@
+            det_time, det_flux, transit_mask, period * 2, epoch - period / 2
+        )
+    )
+    all_features.update(
+        double_period_features(tic, double_fold_time, double_fold_flux, period)
+    )
+
+    half_fold_time, half_fold_flux, _, _ = preprocess.phase_fold_and_sort_light_curve(
+        det_time, det_flux, transit_mask, period / 2, epoch
+    )
+    all_features.update(
+        half_period_features(tic, half_fold_time, half_fold_flux, period, duration)
+    )
+
+    return {k + tag: v for k, v in all_features.items()}, fold_num
+
+
+def prediction_features(
+    tce: pd.Series,
+    time: np.ndarray,
+    flux: np.ndarray,
+    aperture_fluxes: dict[str, tuple[np.ndarray, np.ndarray]],
+    breakspaces: list[Optional[float]] = [0.3, 5.0, None],
+):
+    """
+    Assemble all input features for a lightcurve.
+
+    Params
+    ------
+    tce: pd.Series
+        Row of dataframe containing relevant TCE parameters. Should include "Astro ID", "Per",
+        "Epoc", "Dur", "Depth", "Tmag", "SMass", "SRad", "SRadEst".
+    time: array[float]
+        Array of time values in lightcurve.
+    flux: array[float]
+        Array of flux values in lightcurve.
+    aperture_fluxes: dict[str, tuple[np.ndarray, np.ndarray]]
+        For triage model: empty dict.
+        For vetting model: dict of {aperture_name: (lightcurve_time, lightcurve_flux)} for all
+        apertures to be considered ('s', 'm', 'l').
+    breakspaces: list[Optional[float]]
+        Breakspaces to use in spline detrending. None indicates that various breakspaces should be
+        tried and the best selected. Default = [0.3, 5.0, None].
+
+    Returns
+    -------
+    features: dict[str, float | array[float]]
+        Dict of {feature_name: feature_value} for all inputs for the lightcurve.
+    """
+    all_features = {}
+
+    for breakspace in breakspaces:
+        breakspace_features, fold_num = standard_view_features(
+            tce["Astro ID"],
+            time,
+            flux,
+            tce["Per"],
+            tce["Epoc"],
+            tce["Dur"],
+            breakspace,
+            aperture_fluxes,
+        )
+        all_features.update(breakspace_features)
+
+    scalar_features = {
+        "astro_id": tce["Astro ID"],
+        "Period": tce["Per"],
+        "Duration": tce["Dur"],
+        "Transit_Depth": tce["Depth"],
+        "Tmag": tce["Tmag"],
+        "star_mass": tce["SMass"] if not np.isnan(tce["SMass"]) else 0.0,
+        "star_mass_present": float(np.isnan(tce["SMass"])),
+        "star_rad": tce["SRad"] if not np.isnan(tce["SRad"]) else 0.0,
+        "star_rad_present": float(np.isnan(tce["SRad"])),
+        "star_rad_est": tce["SRadEst"] if not np.isnan(tce["SRadEst"]) else 0.0,
+        "star_rad_est_present": float(np.isnan(tce["SRadEst"])),
+        "n_folds": len(set(fold_num)),
+        "n_points": len(fold_num),
+    }
+    for feature, value in scalar_features.items():
+        assert not np.isnan(
```

Comment:
We'll probably want to raise something other than an assert error. If anyone runs `python -O`, this line will not raise.

Reply:
Use pandas to do a fast check.

Reply:
This is a dictionary, not a dataframe, so pandas can't be faster than iterating.